seismicpro/gather/gather.py

```diff
@@ -1213,14 +1224,16 @@ def resample(self, new_sample_interval, kind=3, enable_anti_aliasing=True):
             filter_size = int(40 * new_sample_interval / current_sample_interval)
             self.bandpass_filter(high=0.9*nyquist_frequency, filter_size=filter_size, window="hann")
 
-        new_samples = np.arange(self.samples[0], self.samples[-1] + 1e-6, new_sample_interval, self.samples.dtype)
+        new_n_samples = int((self.samples[-1] - self.samples[0]) // new_sample_interval) + 1
```

Comment:
To make rouding logic explicit
```suggestion
        new_n_samples = math.floor((self.samples[-1] - self.samples[0]) // new_sample_interval) + 1
```

Reply:
Done